src/Test.UnitTests.Sarif/Core/ToolTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.ObjectModel;
+using FluentAssertions;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Core
+{
+    public class ToolTests
+    {
+        private class DottedQuadFileVersionTestCase
+        {
+            public DottedQuadFileVersionTestCase(string input, string expectedOutput)
+            {
+                Input = input;
+                ExpectedOutput = expectedOutput;
+            }
+
+            public string Input { get; }
+            public string ExpectedOutput { get; }
+        }
+
+        private static ReadOnlyCollection<DottedQuadFileVersionTestCase> s_dottedQuadFileVersionTestCases =
+            new ReadOnlyCollection<DottedQuadFileVersionTestCase>(new DottedQuadFileVersionTestCase[]
+            {
+                new DottedQuadFileVersionTestCase(
```

Comment:
Nit: When I want to test a number of scenarios, I really want them to be one line each so I can read them quickly. You might consider not using named parameters here and formatting each on a single line to read quickly. I also sometimes choose short method names to minimize the "not the test input/output" content on the line.